src/main/java/oof/command/ViewTrackerCommand.java

```diff
@@ -0,0 +1,114 @@
+     * spent on each Module by user in blocks of 10 minutes.
+     *
+     * @param semesterList Instance of SemesterList that stores Semester objects.
+     * @param tasks        Instance of TaskList that stores Task objects.
+     * @param ui           Instance of Ui that is responsible for visual feedback.
+     * @param storage      Instance of Storage that enables the reading and writing of Task
+     *                     objects to hard disk.
+     * @throws OofException if TrackerList is empty.
+     */
+    @Override
+    public void execute(SemesterList semesterList, TaskList tasks, Ui ui, Storage storage) throws OofException {
+        TrackerList trackerList = storage.readTrackerList();
+        ModuleTrackerList moduleTrackerList = timeSpentByModule(trackerList);
+        if (moduleTrackerList.getSize() == EMPTY) {
+            throw new OofException("There are no completed Assignments!");
+        }
+        ModuleTrackerList sortedTL = sortAscending(moduleTrackerList);
+        ui.printTrackerDiagram(sortedTL);
+    }
+
+    /**
+     * Calculate total time spent on various modules.
+     * @param trackerList     list of Tracker objects.
+     * @return          TrackerList of Tracker objects.
+     */
+    private ModuleTrackerList timeSpentByModule(TrackerList trackerList) {
+        ModuleTrackerList moduleTrackerList = new ModuleTrackerList();
+        Tracker tracker;
+
+        for (int i = 0; i < trackerList.getSize(); i++) {
+            tracker = trackerList.getTracker(i);
+            String moduleCode = tracker.getModuleCode();
+            boolean isFound = false;
+
+            if (moduleTrackerList.getSize() == EMPTY) {
+                long timeTaken = tracker.getTimeTaken();
+                ModuleTracker moduleTracker = new ModuleTracker(moduleCode,timeTaken);
+                moduleTrackerList.addModuleTracker(moduleTracker);
+            } else {
+                for (int j = 0; j < moduleTrackerList.getSize(); j++) {
+                    ModuleTracker moduleTracker = moduleTrackerList.getModuleTracker(j);
+                    String savedModule = moduleTracker.getModuleCode();
+                    if (moduleCode.equals(savedModule)) {
+                        long timeTaken = tracker.getTimeTaken();
+                        long totalTime = moduleTracker.getTimeTaken();
+                        totalTime += timeTaken;
+                        moduleTracker.setTimeTaken(totalTime);
+                        isFound = true;
+                        break;
+                    }
+                }
+                if (!isFound) {
+                    long timeTaken = tracker.getTimeTaken();
+                    ModuleTracker moduleTracker = new ModuleTracker(moduleCode, timeTaken);
+                    moduleTrackerList.addModuleTracker(moduleTracker);
+                }
+            }
+        }
+        return moduleTrackerList;
+    }
```

Comment:
Arrow head.

Reply:
The code has been rectified and now has methods where appropriate.